src/core/libmaven/mavenparameters.cpp

```diff
@@ -232,8 +232,15 @@ void MavenParameters::setOptionsDialogSettings(const char* key, const char* valu
 {
     mavenSettings[const_cast<char*>(key)] = const_cast<char*>(value);
 
-    if(strcmp(key, "ionizationMode") == 0)
-        ionizationMode = atof(value);
+    if(strcmp(key, "ionizationMode") == 0){
+        int indexOfIonizationMode=*value-'0';
+        /**
+         * map index of ionization-mode to value (-1,0,1) of ionizationMode
+         */
+        if(indexOfIonizationMode==1) ionizationMode=0;
```

Comment:
MainWindow::setTotalCharge() does the same thing. Check that function out.

Reply:
@shubhra-agrawal  we still need to do it here. if we don't then, every call to this function will change the ionizationMode to hold the index and not the actual value

Reply:
@rish9511 Yes, we need to do it here. I merely suggested using the already defined function here if possible. Short answer is, we cannot. So this is fine.